src/main/resources/templates/db_scripts/create_note_type_view.sql

```diff
@@ -0,0 +1,13 @@
+-- Custom script to create note_type_view. Changes in this file will not result in an update of the view.
+-- To change the view, update this script and copy it to the appropriate scripts.snippet field of the schema.json
+CREATE OR REPLACE VIEW note_type_view AS
+   SELECT note_type.id,
+    jsonb_build_object(
+    'id', note_type.jsonb ->>'id',
+    'name', note_type.jsonb ->> 'name',
+    'usage', json_build_object('noteTotal', count(note_data.jsonb ->> 'id')),
+    'metadata', note_type.jsonb -> 'metadata')
+    AS jsonb
+   FROM note_type
+     LEFT JOIN note_data ON (note_data.jsonb ->> 'typeId'::text) = (note_type.jsonb ->> 'id'::text)
```

Comment:
could be simplified to ` LEFT JOIN note_data ON (note_data.jsonb ->> 'typeId')::uuid = (note_type.id)` 
as `typeId` field is a uuid of note type.